tools/shared/src/main/scala/org/scalajs/core/tools/javascript/JSDesugaring.scala

```diff
@@ -117,28 +118,44 @@ private[javascript] object JSDesugaring {
    */
   private[javascript] def desugarToFunction(
       classEmitter: ScalaJSClassEmitter, enclosingClassName: String,
-      params: List[ParamDef], body: Tree, isStat: Boolean)(
+      params: List[ParamDef], body: Tree, isStat: Boolean,
+      enclosingMethodName: String, isStatic: Boolean)(
```

Comment:
I suggest putting `enclosingMethodName` and `isStatic` right after `enclosingClassName`, as those 3 go together. Same for the other two `desugarTo` methods.